config-linux.md

```diff
@@ -35,11 +35,17 @@ The following parameters can be specified to setup namespaces:
     * **`user`** the container will be able to remap user and group IDs from the host to local users and groups within the container.
     * **`cgroup`** the container will have an isolated view of the cgroup hierarchy.
 
-* **`path`** *(string, OPTIONAL)* - an absolute path to namespace file in the [runtime mount namespace](glossary.md#runtime-namespace)
+* **`path`** *(string, OPTIONAL)* - an absolute path to namespace file in the [runtime mount namespace](glossary.md#runtime-namespace).
+    The runtime MUST place the container process in the namespace associated with that `path`.
+    The runtime MUST [generate an error](runtime.md#errors) if `path` is not associated with a namespace of type `type`.
+
+    If `path` is not specified, the runtime MUST create a new [container namespace](glossary.md#container-namespace) of type `type`.
+    For hierarchical namespaces (e.g. `pid`, `user`), the new container namespace MUST be a child of the [runtime namespace](glossary.md#runtime-namespace) of that type.
+    For seeded namespaces (e.g. `mount`, `uts`), the new container namespace MUST be seeded by the runtime namespace of that type.
```

Comment:
> On the one hand they are useful for some circumstances…

For example, if you ask for a new `uts` namespace but do not set the optional [`hostname`][1], having the seed defined means that the hostname in the container UTS namespace is well-defined (it will be whatever the hostname was in the runtime UTS namespace).

This is less of an issue for the `mount` namespace, because with [`root.path`] REQUIRED, there's no way to avoid clobbering whatever mounts you got from your seed (which makes *not* asking for a new `mount` namespace exciting ;).

> … but I'm having trouble convincing myself that these requirements are not going make weird restrictions for a non-cli interface…

We already have some of “runtime namespace” conditions (e.g. for [`linux.namespaces[].path`][3] which have this “caller may not have a good idea what the runtime namespace actually is” situation.  And this is clearly an issue already for namespaces which are *not* set in [`linux.namespaces[]`][4].

I expect runtimes, CLI and otherwise, will document what they consider the runtime namespaces to be to avoid any ambiguity.  For a CLI, it would probably, but not necessarily, be the caller's namespaces.  For a web UI, it might be:

> The runtime mount namespace will only contain a root directory owned by 0:0 with 755 permissions.  The runtime UTS namespace will have a `test` hostname.  …

But I don't see how this has an impact unique to the “defining the seed namespace” use case.

[1]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc5/config.md#hostname
[2]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc5/config.md#root
[3]: https://github.com/opencontainers/runtime-spec/blame/v1.0.0-rc5/config-linux.md#L38
[4]: https://github.com/opencontainers/runtime-spec/blame/v1.0.0-rc5/config-linux.md#L41